app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -101,23 +105,59 @@ class AccountController @Inject() (
-      Ok(
-        leadAccountPage(
-          undertaking,
-          undertaking.getAllNonLeadEORIs().nonEmpty,
-          isTimeToReport,
-          dueDate,
-          isOverdue
-        )
-      )
-    } else Ok(nonLeadAccountPage(undertaking))
+    val isOverdue = ReportReminderHelpers.isOverdue(undertaking.lastSubsidyUsageUpdt, currentDay)
+
+    if (undertaking.isLeadEORI(eori)) {
+      val result = for {
+        nilReturnJourney <- store
+          .get[NilReturnJourney]
+          .toContext
+          .orElse(store.put[NilReturnJourney](NilReturnJourney()).toContext)
+        updatedJourney <-
+          if (isUpdateNeededForNilJourneyCounter(nilReturnJourney))
+            store.update[NilReturnJourney](updateNilReturnCounter).toContext
+          else nilReturnJourney.some.toContext
+        result <- Ok(
+          leadAccountPage(
+            undertaking,
+            undertaking.getAllNonLeadEORIs().nonEmpty,
+            isTimeToReport,
+            dueDate,
+            isOverdue,
+            isNilReturnDoneRecently(updatedJourney),
+            currentDay.plusDays(dueDays).toDisplayFormat
+          )
+        ).toFuture.toContext
+      } yield result
+
+      result.getOrElse(handleMissingSessionData("Nil Return Journey"))
+
+    } else Ok(nonLeadAccountPage(undertaking)).toFuture
   }
 
+  private def updateNilReturnJourney(nrj: Option[NilReturnJourney])(f: NilReturnJourney => NilReturnJourney) =
+    nrj.map(f)
+
+  private def updateNilReturnCounter(nrj: Option[NilReturnJourney]) = updateNilReturnJourney(nrj) { nj =>
+    nj.copy(nilReturnCounter = nj.nilReturnCounter + 1)
+  }
+
+  //nilReturnCounter is used to track when user has moved to home account page after nil return claim
+  //This counter also helps in identifying when to display the success message which should be displayed only once
+  //By default,  NilReturnJourney has counter set to 0. When user submit on No claim page, the count is set to 1, indicating user has started the nil return journey.
+  // Home account has logic to update the counter only if it's 1 or 2. Since the counter is 1 ,when the user is redirected to home account, counter get updated to 2.
+  // Home page has logic to display the message only if the counter is 2 . At this point the message will be displayed.
+  //If user refreshes or return to home page via another journey, counter is updated to 3 and success message is no longer displayed
+  //Since the counter is 3 now, it will no longer be updated because of this func logic and will be rest to 1 if user goes on to nil return journey again.
+  private def isUpdateNeededForNilJourneyCounter(nilReturnJourney: NilReturnJourney) =
+    nilReturnJourney.nilReturnCounter == 1 || nilReturnJourney.nilReturnCounter == 2
+
+  //Logic to check if the success message be displayed on the Home account page
+  private def isNilReturnDoneRecently(nilReturnJourney: NilReturnJourney) = nilReturnJourney.nilReturnCounter == 2
```

Comment:
NABD - but you could have functions on the journey which expose this with a helpful name. I.e.e `nilReturnJourney.recentNilReturnDone` for 2, and ``nilreturnJourney.userStartedNilReturnJourney` for 1. 

NABD though.